lib/array_equals.rb

```diff
@@ -1,5 +1,21 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == [] && array2 == [] || array1 == nil && array2 == nil
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  end
+
+  if array1.count == array2.count
+    check = true
+    array1.count.times do |i|
+      if array1[i] != array2[i]
+        check = false
```

Comment:
Super minor but you could get rid of the check variable and instead return False on line 14, which would allow the code to return early instead of iterating over all array elements. You could then return True on line 17, as that would mean that the array elements were found to be matches.